### ISSUE TITLE
 Fails to connect using Yubikey

### ISSUE THREAD

**Issue by Samega7Cattac** [microsoft/vscode-remote-release#6688]
- VSCode Version: 1.66.2
- Local OS Version: Arch Linux
- Remote OS Version: Windows 11
- Remote Extension/Connection Type: SSH
- Logs:
```
[21:33:36.259] Log Level: 2
[21:33:36.261] remote-ssh@0.79.2022040715
[21:33:36.262] linux x64
[21:33:36.268] SSH Resolver called for "ssh-remote+win11", attempt 1
[21:33:36.269] "remote.SSH.useLocalServer": true
[21:33:36.269] "remote.SSH.path": undefined
[21:33:36.270] "remote.SSH.configFile": /home/samega7cattac/.ssh/config
[21:33:36.270] "remote.SSH.useFlock": true
[21:33:36.270] "remote.SSH.lockfilesInTmp": false
[21:33:36.271] "remote.SSH.localServerDownload": auto
[21:33:36.271] "remote.SSH.remoteServerListenOnSocket": false
[21:33:36.272] "remote.SSH.showLoginTerminal": true
[21:33:36.272] "remote.SSH.defaultExtensions": []
[21:33:36.272] "remote.SSH.loglevel": 2
[21:33:36.273] "remote.SSH.enableDynamicForwarding": true
[21:33:36.273] "remote.SSH.enableRemoteCommand": true
[21:33:36.274] "remote.SSH.serverPickPortsFromRange": {}
[21:33:36.274] "remote.SSH.serverInstallPath": {}
[21:33:36.304] SSH Resolver called for host: win11
[21:33:36.304] Setting up SSH remote "win11"
[21:33:36.307] Acquiring local install lock: /tmp/vscode-remote-ssh-d761b3ab-install.lock
[21:33:36.309] Looking for existing server data file at /home/samega7cattac/.config/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-d761b3ab-dfd34e8260c270da74b5c2d86d61aee4b6d56977-0.79.2022040715/data.json
[21:33:36.310] Using commit id "dfd34e8260c270da74b5c2d86d61aee4b6d56977" and quality "stable" for server
[21:33:36.315] Install and start server if needed
[21:33:36.332] PATH: /opt/devkitpro/tools/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/android-sdk/platform-tools:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
[21:33:36.332] Checking ssh with "ssh -V"
[21:33:36.341] > OpenSSH_9.0p1, OpenSSL 1.1.1o  3 May 2022

[21:33:36.345] Using SSH config file "/home/samega7cattac/.ssh/config"
[21:33:36.346] askpass server listening on /run/user/1000/vscode-ssh-askpass-64c18dfef9d9a8c08af600c7f415d4defa5ad6ae.sock
[21:33:36.347] Spawning local server with {"serverId":1,"ipcHandlePath":"/run/user/1000/vscode-ssh-askpass-79c75c8584e411406270f796b2fd59c334f5003a.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","33063","-o","ConnectTimeout=15","-F","/home/samega7cattac/.ssh/config","win11"],"serverDataFolderName":".vscode-server","dataFilePath":"/home/samega7cattac/.config/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-d761b3ab-dfd34e8260c270da74b5c2d86d61aee4b6d56977-0.79.2022040715/data.json"}
[21:33:36.347] Local server env: {"SHELL":"/bin/bash","DISPLAY":":1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/opt/visual-studio-code/code","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"--ms-enable-electron-run-as-node","VSCODE_SSH_ASKPASS_MAIN":"/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/run/user/1000/vscode-ssh-askpass-64c18dfef9d9a8c08af600c7f415d4defa5ad6ae.sock"}
[21:33:36.355] Spawned 50938
[21:33:36.506] > local-server-1> Spawned ssh, pid=50946
[21:33:36.513] stderr> OpenSSH_9.0p1, OpenSSL 1.1.1o  3 May 2022
[21:33:36.756] stderr> debug1: Server host key: ssh-ed25519 SHA256:FTbiNavK2dDUlzW5L8IQO81eskT+A/ozEtq4rqInUaU
[21:33:36.851] stderr> 7Catt@192.168.1.139: Permission denied (publickey,keyboard-interactive).
[21:33:36.852] > local-server-1> ssh child died, shutting down
[21:33:36.858] Server delay-shutdown request failed: connect ENOENT /run/user/1000/vscode-ssh-askpass-79c75c8584e411406270f796b2fd59c334f5003a.sock
[21:33:36.858] Local server exit: 0
[21:33:36.858] Received install output: local-server-1> Spawned ssh, pid=50946
OpenSSH_9.0p1, OpenSSL 1.1.1o  3 May 2022
debug1: Server host key: ssh-ed25519 SHA256:FTbiNavK2dDUlzW5L8IQO81eskT+A/ozEtq4rqInUaU
7Catt@192.168.1.139: Permission denied (publickey,keyboard-interactive).
local-server-1> ssh child died, shutting down

[21:33:36.860] Resolver error: Error: Permission denied (publickey,keyboard-interactive).
	at Function.Create (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:585789)
	at /home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:583840
	at Object.t.handleInstallOutput (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:584363)
	at Object.e [as tryInstallWithLocalServer] (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:624908)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async /home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:643945
	at async Object.t.withShowDetailsEvent (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:647692)
	at async /home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:623380
	at async b (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:619886)
	at async Object.t.resolveWithLocalServer (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:622995)
	at async Object.t.resolve (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:645302)
	at async /home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.79.2022040715/out/extension.js:1:722235
[21:33:36.865] ------
```

Steps to Reproduce:

1. Setup a ssh connection using a yubikey
2. Try to connect on VSCode

Does this issue occur when you try this locally?: No (I mean I can connect to it using ssh on a terminal)
Does this issue occur when you try this locally and all extensions are disabled?: N/A

**Comment by Samega7Cattac** [comment]
Using those ssh settings on a terminal works.
I'm using the GnuPGP ssh agent. (the SSH_AUTH_SOCK is correctly set for that)

**Comment by tanhakabir** [comment]
I believe this is a symptom of https://github.com/microsoft/vscode-remote-release/issues/4747

What happens if you try with `remote.SSH.useLocalServer: false`?

**Comment by Samega7Cattac** [comment]
I set the useLocalServer to false on the GUI settings and still same issue
```
[20:31:47.840] Log Level: 2
[20:31:47.843] remote-ssh@0.81.2022051115
[20:31:47.843] linux x64
[20:31:47.856] SSH Resolver called for "ssh-remote+win11", attempt 1
[20:31:47.857] "remote.SSH.useLocalServer": false
[20:31:47.857] "remote.SSH.showLoginTerminal": true
[20:31:47.857] "remote.SSH.remotePlatform": {}
[20:31:47.857] "remote.SSH.path": undefined
[20:31:47.857] "remote.SSH.configFile": /home/samega7cattac/.ssh/config
[20:31:47.858] "remote.SSH.useFlock": true
[20:31:47.858] "remote.SSH.lockfilesInTmp": false
[20:31:47.858] "remote.SSH.localServerDownload": auto
[20:31:47.858] "remote.SSH.remoteServerListenOnSocket": false
[20:31:47.858] "remote.SSH.showLoginTerminal": true
[20:31:47.858] "remote.SSH.defaultExtensions": []
[20:31:47.858] "remote.SSH.loglevel": 2
[20:31:47.859] "remote.SSH.enableDynamicForwarding": true
[20:31:47.859] "remote.SSH.enableRemoteCommand": true
[20:31:47.859] "remote.SSH.serverPickPortsFromRange": {}
[20:31:47.859] "remote.SSH.serverInstallPath": {}
[20:31:47.888] SSH Resolver called for host: win11
[20:31:47.889] Setting up SSH remote "win11"
[20:31:47.903] Using commit id "da15b6fd3ef856477bf6f4fb29ba1b7af717770d" and quality "stable" for server
[20:31:47.911] Install and start server if needed
[20:31:49.525] PATH: /opt/devkitpro/tools/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/android-sdk/platform-tools:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
[20:31:49.525] Checking ssh with "ssh -V"
[20:31:49.534] > OpenSSH_9.0p1, OpenSSL 1.1.1o  3 May 2022

[20:31:49.538] Using SSH config file "/home/samega7cattac/.ssh/config"
[20:31:49.539] Remote command length: 6205/8192 characters
[20:31:49.539] Running script with connection command: ssh -T -D 36185 -o ConnectTimeout=15 -F '/home/samega7cattac/.ssh/config' win11 powershell
[20:31:50.355] > 7Catt@192.168.1.139: Permission denied (publickey,keyboard-interactive).
[20:31:50.356] Got some output, clearing connection timeout
[20:31:50.627] "install" terminal command done
[20:31:50.627] Install terminal quit with output: 7Catt@192.168.1.139: Permission denied (publickey,keyboard-interactive).
[20:31:50.627] Received install output: 7Catt@192.168.1.139: Permission denied (publickey,keyboard-interactive).
[20:31:50.629] Resolver error: Error: Permission denied (publickey,keyboard-interactive).
	at Function.Create (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.81.2022051115/out/extension.js:1:585789)
	at /home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.81.2022051115/out/extension.js:1:583840
	at Object.t.handleInstallOutput (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.81.2022051115/out/extension.js:1:584363)
	at Object.t.tryInstall (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.81.2022051115/out/extension.js:1:681708)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async /home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.81.2022051115/out/extension.js:1:644348
	at async Object.t.withShowDetailsEvent (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.81.2022051115/out/extension.js:1:647693)
	at async Object.t.resolve (/home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.81.2022051115/out/extension.js:1:645427)
	at async /home/samega7cattac/.vscode/extensions/ms-vscode-remote.remote-ssh-0.81.2022051115/out/extension.js:1:722519
[20:31:50.638] ------
```

**Comment by tanhakabir** [comment]
Hm okay yeah I think you're blocked by #4747. I'll work on a fix for that in the upcoming days and then you can try again when I have a fix for that out.

Closing as a duplicate of #4747
